File: feature-main/src/main/java/com/smedialink/feature_main/di/MainActivityComponent.java
package com.smedialink.feature_main.di;

import com.smedialink.feature_main.view.MainActivity;

import dagger.BindsInstance;
import dagger.Component;
import home.oleg.placesnearme.scopes.FeatureScope;

@FeatureScope
@Component(modules = MainActivityModule.class)
public interface MainActivityComponent {

    void inject(MainActivity target);

    @Component.Builder
    interface Builder {
        @BindsInstance
        MainActivityComponent.Builder bind(MainActivity target);

        MainActivityComponent build();
    }

    final class Injector {
        private Injector() {
        }

        public static void inject(MainActivity activity) {
            DaggerMainActivityComponent.builder()
                    .bind(activity)
                    .build()
                    .inject(activity);
        }
    }
}

File: feature-main/src/main/java/com/oleg/placesnearme/feature_main/di/MainActivityComponent.kt
package com.oleg.placesnearme.feature_main.di

import com.oleg.placesnearme.feature_main.view.MainActivity

import dagger.BindsInstance
import dagger.Component
import home.oleg.placesnearme.coredi.scopes.FeatureScope

@FeatureScope
@Component(modules = [MainActivityModule::class])
interface MainActivityComponent {

    fun inject(target: MainActivity)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun bind(target: MainActivity): MainActivityComponent.Builder

        fun build(): MainActivityComponent
    }

    object Injector {

        fun inject(activity: MainActivity) {
            DaggerMainActivityComponent.builder()
                    .bind(activity)
                    .build()
                    .inject(activity)
        }
    }
}
